fraud/registry.go

```diff
@@ -0,0 +1,29 @@
+package fraud
+
+import (
+	"fmt"
+	"sync"
+)
+
+var unmarshalersLk = sync.RWMutex{}
+var defaultUnmarshalers = map[ProofType]ProofUnmarshaler{}
+
+func init() {
+	Register(BadEncoding, UnmarshalBEFP)
+}
```

Comment:
Let's move it close to the definition of the BEFP itself. Technically, now, BEFP can now live outside of this package - in DASer :) (but that does not matter too much)

Reply:
I suggest to keep it behind ` fraud.NewService`. wdyt? @Wondertan